subsys/bluetooth/services/ias.c

```diff
@@ -0,0 +1,122 @@
+/** @file
+ *  @brief IAS Service sample
+ */
+
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bluetooth/att.h"
+#include "net/buf.h"
+#include "bluetooth/services/ias.h"
+#include <init.h>
+#include <stdint.h>
+#include <zephyr.h>
+#include <logging/log.h>
+#include <bluetooth/gatt.h>
+
+#define LOG_LEVEL CONFIG_BT_IAS_LOG_LEVEL
+LOG_MODULE_REGISTER(ias);
+
+#define BT_IAS_ALERT_MAX_LEN 1
+
+typedef enum __packed {
+	BT_IAS_ALERT_LVL_NO_ALERT,
+	BT_IAS_ALERT_LVL_MILD_ALERT,
+	BT_IAS_ALERT_LVL_HIGH_ALERT,
+} bt_ias_alert_lvls;
+
+#define GATT_PERM_WRITE_MASK    (BT_GATT_PERM_WRITE | \
+				 BT_GATT_PERM_WRITE_ENCRYPT | \
+				 BT_GATT_PERM_WRITE_AUTHEN)
+
+#if defined(CONFIG_BT_IAS_DEFAULT_PERM_W_AUTHEN)
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE_AUTHEN
+#elif defined(CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT)
+#define IAS_GATT_PERM_DEFUALT CONFIG_BT_IAS_DEFAULT_PERM_W_ENCRYPT 
+#else
+#define IAS_GATT_PERM_DEFAULT BT_GATT_PERM_WRITE
+#endif
+
+bt_ias_alert_lvls g_alert_level;
+
+static void alert_level(bt_ias_alert_lvls alert_lvl)
+{
+	if (alert_lvl == g_alert_level) {
```

Comment:
What if the ongoing alert has been stopped by the user?  In that case, this should start a new one, rather than just returning.